flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/runtime/aggregate/AggregateMapFunction.scala

```diff
@@ -44,7 +44,7 @@ class AggregateMapFunction[IN, OUT](
 
   override def map(value: IN): OUT = {
     
-    val input = value.asInstanceOf[Row]
+    val input = value.asInstanceOf[Product]
```

Comment:
In `Execution mode = COLLECTION, Table config = EFFICIENT` for `testUnion`, the `value` is of `scala.Tuple3` type, not work as expected?